concepts/patterns/introduction.md

```diff
@@ -0,0 +1,75 @@
+# Introduction
+
+Recall from the Fundamentals concept, an AWK program is composed of **pattern-action pairs**.
+
+```awk
+pattern1 { action1 }
+pattern2 { action2 }
+...
+```
+
+## What do we mean by "pattern"?
+
+The "pattern" is any AWK expression that results in a true or false value.
+
+## The Empty Pattern
+
+The pattern can be omitted.
+In this case, the action is performed for every record.
+
+We can print all the usernames in the passwd file.
+
+```sh
+awk -F: '{print $1}' /etc/passwd
+```
+
+## Regular Expressions
+
+AWK can do what grep can do.
+
+This example compares the whole record (`$0`) to the regex.
+
+```sh
+awk '/regex/' data.txt
+```
+
+A regular expression literal is enclosed in `/` slashes.
+
+Use the `~` regex-matching operator to match a particular field. 
+To find the users who use bash:
+
+```sh
+awk -F: '$7 ~ /bash/ {print $1}' /etc/passwd
+```
+
+`!~` is the "regex does **not** match" operator.
+
+We'll get more into GNU AWK's regular expression flavour in a further concept.
+
+
+## Arithmetic expressions
+
+Arithmetic expressions can be used as patterns.
```

Comment:
This is a boolean expression :) An arithmetic expression would be  `$3 + 5` for example, which leads into the truthiness of numbers.

Reply:
Hmm, I can't think of something much better.

Reply:
Rename "Arithmetic expressions" to "Boolean expressions"? Or, rename the section to "Arithmetic":

```
## Boolean expressions

...

## Arithmetic operators

Expressions can be any valid AWK. For instance, they can contain arithmetic and boolean operators:

```sh
awk '
    $1 * 2 > max { print("The max value has doubled! Oh my!!"); }
    $1 > max { max = $1 }
'
```

Reply:
Added a different suggestion: https://github.com/exercism/awk/pull/209/files#r1284039678

Reply:
If we want to get really pedantic...

This is arguably not a Boolean expression, since the definition of "a Boolean expression" in computer science is any expression that evaluates to a Boolean value, and AWK only has 1 and 0. It does fill the role though...

This is also technically not Arithmetic, comparison is Algebra :)

Anyway, such distinctions are not actually useful in a brief introduction to AWK; it's broadly understood what `x >= 1000` _means_ and the terminology is not important for an overview. As such I suggest to just say "Expressions" and "AWK Expressions" and move on